articles/how-to-consume-data-from-an-API-using-react/index.md

```diff
@@ -0,0 +1,424 @@
+const [search, setSearch] = useState('');
+```
+
+You can also have a constant for your API key or include it in your url directly.
+
+```JSX
+const API_KEY = "your_api_key";
+const url = `http://www.omdbapi.com/?t=${search}&apikey=${API_KEY}`;
+```
+
+The `App` component will also has several functions. They are:
+
+#### 1.	getMovie
+
+This method will be called whenever the search button is clicked. It will make a request to the API and return a movie object.
+
+A [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) is a proxy for a value not necessarily known when the promise is created. It allows you to associate handlers with an asynchronous action's eventual success value or failure reason. This lets asynchronous methods return values like synchronous methods: instead of immediately returning the final value, the asynchronous method returns a promise to supply the value at some point in the future. 
+
+A Promise can either be in one of the following states:
+
+- Pending: It means that the Promise has neither been rejected or completed.
+
+- Fulfilled: It means the operation was successful.
+
+- Rejected: This state means that the operation failed.
+
+When a Promise is pending, it can either be completed successfully or rejected. The `then` method helps handle any of the resulting state.
+
+Here is an example of handling a promise:
+
+```JSX
+const dataPromise = (new Promise(fetchData))
+  .then(handleData1)
+  .then(handleData1)
+  .catch(handleRejectedAny);
+```
+
+As shown above, we handle the Promise when it is rejected in the catch.
+
+The `getMovie` is a async function. An async function is a function declared with the async keyword. Async functions are instances of the AsyncFunction constructor, and the await keyword is permitted within them.
+
+```JSX
+async getData(){
+  //do something
+  //await statement
+}
+```
+
+This is because network operations are asynchronous in nature. Therefore, an API request will return a promise and we should wait for the promise to get resolved. The promise will be resolved if the API request is successful. If not, the promise will be rejected.
+
+The `await` keyword allows you to wait for a Promise to get resolved. We can incorporate the await keyword in the getData method as follows.
+
+```JSX
+async getData(){
+    var user = await fetch("your_url")
+}
+```
+
+You can learn more about async, await, and promise from [here](https://www.w3schools.com/js/js_promise.asp). This tutorial uses async and await methods.
+
+We can handle server and network errors using a `try/catch` block. We send a network request inside the `try` block . If an error occurs, the catch block will be executed.
+
+Here is the code for the `getMovie` function.
+
+```JSX
+const getMovie = async () => {
+  try {
+     const response = await fetch(url);
+      const data = await response.json()
+      setMovie(data);
+   } catch (e) {
+       console.error(e.toString);
+   } 
+}
```

Comment:
Inconsistent code snippet: Here, you've provided `e.toString` in the catch block, and below, you've just provided `e`. 

Just print `e` in both places. Also, rename the variable to `error` to provide better clarity.